app/Community/Actions/UpdateGameClaimAction.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Actions;
+
+use App\Community\Enums\ArticleType;
+use App\Community\Enums\AwardType;
+use App\Community\Enums\ClaimSetType;
+use App\Community\Enums\ClaimStatus;
+use App\Community\Enums\ClaimType;
+use App\Community\Requests\UpdateGameClaimRequest;
+use App\Models\AchievementSetClaim;
+use App\Models\Game;
+use App\Models\PlayerBadge;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Auth;
+
+class UpdateGameClaimAction
+{
+    public function execute(AchievementSetClaim $claim, UpdateGameClaimRequest $request): void
+    {
+        /** @var User $currentUser */
+        $currentUser = Auth::user();
+
+        $newValues = $request->validated();
+
+        $auditMessage = "{$currentUser->display_name} updated {$claim->user->display_name}'s claim.";
+
+        if (array_key_exists('type', $newValues)) {
+            $newType = (int) $newValues['type'];
```

Comment:
It may be safer to lift up `int` casts to `UpdateGameClaimRequest`. This way, if the code ever gets refactored, whoever uses the request doesn't need to remember to manually perform the cast.

```php
// in UpdateGameClaimRequest.php

public function rules(): array
{
    // ... stuff here ...
}

protected function passedValidation(): void
{
    $this->merge([
        'status' => $this->status ? (int) $this->status : null,
        'special' => $this->special ? (int) $this->special : null,
        'type' => $this->type ? (int) $this->type : null,
        'set_type' => $this->set_type ? (int) $this->set_type : null,
    ]);
}
```